### PR TITLE
[BuildTasks] Fix resolution for nested types

### LINKED ISSUE #16293: [.NET 8 Preview 6] Static extension can't resolve nested class references from class libraries
### Description

When referencing nested static values from a class library in Xaml using the StaticExtension, Visual Studio complains about being unable to resolve the type:

```
Error	XFC0000	Cannot resolve type "clr-namespace:StaticExtensionApp.ClassLibrary;assembly=StaticExtensionApp.ClassLibrary:LibraryConstants+LibraryIconCodes".	StaticExtensionApp	C:\Users\bross\source\repos\StaticExtensionApp\StaticExtensionApp\MainPage.xaml	47	
```

Here's my `dotnet workload list` output:
```
Installed Workload Id      Manifest Version                            Installation Source
-----------------------------------------------------------------------------------------------------------
maui-ios                   8.0.0-preview.6.8686/8.0.100-preview.6      VS 17.6.33829.357, VS 17.7.33913.275
maui-android               8.0.0-preview.6.8686/8.0.100-preview.6      VS 17.6.33829.357, VS 17.7.33913.275
maccatalyst                16.4.8646-net8-p6/8.0.100-preview.6         VS 17.6.33829.357, VS 17.7.33913.275
maui-windows               8.0.0-preview.6.8686/8.0.100-preview.6      VS 17.6.33829.357, VS 17.7.33913.275
maui-maccatalyst           8.0.0-preview.6.8686/8.0.100-preview.6      VS 17.6.33829.357, VS 17.7.33913.275
ios                        16.4.8646-net8-p6/8.0.100-preview.6         VS 17.6.33829.357, VS 17.7.33913.275
android                    34.0.0-preview.6.359/8.0.100-preview.6      VS 17.6.33829.357, VS 17.7.33913.275
```

### Steps to Reproduce

1. Clone my repository
2. Attempt to build and run the app

### Link to public reproduction project repository

https://github.com/mavispuford/StaticExtensionsAppIssue

### Version with bug

Unknown/Other

### Last version that worked well

7.0.86

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Android 13, Windows 10.0.19044 Build 19044

### Did you find any workaround?

No

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by pjcollins**
Fixes #16293
Context: https://github.com/dotnet/maui/commit/db74ff7411d20524180d13f07805fc1a574e1c44

The `XamlC` task was failing to resolve public nested types from a class
library.  Fix this by updating the `IsPublicOrVisibleInternal` checks
added in commit db74ff74 to also accept public nested types.

### CODE REVIEW

File: src/Controls/tests/Xaml.UnitTests.ExternalAssembly/Gh16923Library.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+namespace Microsoft.Maui.Controls.ControlGallery
+{
+	public static class Gh16923Library
+	{
+		public const string LibraryConstant = nameof(LibraryConstant);
+
+		public static class NestedIconCodes
+		{
+			public const string LibraryIconCode = nameof(LibraryIconCode);
```
Comment (jonathanpeppers) [inline src/Controls/tests/Xaml.UnitTests.ExternalAssembly/Gh16923Library.cs:9]: Should we test an `[InternalsVisibleTo]` nested class, too? (not sure if we have a class library setup for that yet)

**Review by jonathanpeppers — DISMISSED** [review]

File: src/Controls/tests/Xaml.UnitTests.ExternalAssembly/Gh16923Library.cs (hunk 1 above)
Comment (pjcollins) [reply src/Controls/tests/Xaml.UnitTests.ExternalAssembly/Gh16923Library.cs]: Thanks, I've simplified the xaml in the test a bit and added an internal nested class to the test.

**Review by jonathanpeppers — APPROVED** [review]

**Comment by pjcollins** [comment]
@rmarinho looks like some known UI test issues may be blocking this? What is the best path forward to resolve this?

**Comment by pjcollins** [comment]
> We can ignore the failing uitests.

Thanks, do we need any other reviewers on this or could an owner override those check requirements and merge this?

**Review by rmarinho — APPROVED** [review]

**Pull request merged**